extensions/replication/queueProcessor/QueueProcessor.js

```diff
@@ -200,6 +205,30 @@ class QueueProcessor extends EventEmitter {
         this.accountCredsCache = {};
     }
 
+    _setupRedis() {
+        // redis subscribe to site specific channel
+        const channelName = `${this.repConfig.topic}-${this.site}`;
+        this._redis.subscribe(channelName, err => {
+            if (err) {
+                this.logger.fatal('queue processor failed to subscribe to ' +
+                                  `crr redis channel for site ${this.site}`,
+                                  { method: 'QueueProcessor.constructor',
+                                    error: err });
+                process.exit(1);
+            }
+            this._redis.on('message', (channel, message) => {
+                if (channel === channelName && this._consumer[message]) {
```

Comment:
I guess you cannot do the economy here of having to dispatch explicitly `"pauseService" -> consumer.pause()` and `"resumeService" -> consumer.resume()`, as those are part of the implementation logic (i.e. to pause my service, I have to pause that consumer, but I also may want later to add some more pausing logic or feedback, like logging).

If you prefer you can have the `pauseService` and `resumeService` methods defined in the `QueueProcessor` class and dispatch by accessing `this[message]` but I feel this will be more confusing than explicit method calls, and also you still have to check for validity of the message.

Reply:
If adding more pausing logic, I think implementing a separate `pauseService` or `resumeService` method in QueueProcessor would make sense because we will not only be stopping the consumer but performing other actions as well.

I guess if we wanted to handle messages explicitly, I can match the message by the desired or expected commands using a case/switch or if/else statements? Then these can use the appropriate methods in QueueProcessor to then handle the actions necessary (i.e. logging, unsubscribing consumer, etc)

Reply:
That, or a mapping object that maps the message action names with the appropriate handler method, either works! I tend to prefer slightly the mapping object which feels cleaner in the code.